components/gitpod-cli/cmd/timeout-extend.go

```diff
@@ -19,27 +20,29 @@ import (
 var extendTimeoutCmd = &cobra.Command{
 	Use:   "extend",
 	Short: "Extend timeout of current workspace",
-	Run: func(cmd *cobra.Command, args []string) {
-		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx, cancel := context.WithTimeout(cmd.Context(), 5*time.Second)
 		defer cancel()
 		wsInfo, err := gitpod.GetWSInfo(ctx)
 		if err != nil {
-			fail(err.Error())
+			return err
 		}
 		client, err := gitpod.ConnectToServer(ctx, wsInfo, []string{
 			"function:setWorkspaceTimeout",
 			"resource:workspace::" + wsInfo.WorkspaceId + "::get/update",
 		})
 		if err != nil {
-			fail(err.Error())
+			return err
 		}
+		defer client.Close()
 		if _, err := client.SetWorkspaceTimeout(ctx, wsInfo.WorkspaceId, time.Minute*180); err != nil {
 			if err, ok := err.(*jsonrpc2.Error); ok && err.Code == serverapi.PLAN_PROFESSIONAL_REQUIRED {
-				fail("Cannot extend workspace timeout for current plan, please upgrade your plan")
+				return GpError{OutCome: utils.Outcome_Success, Message: "Cannot extend workspace timeout for current plan, please upgrade your plan"}
```

Comment:
It should be an user error and there should be an error code like `plan_upgrade_required`